org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/EventFieldEditingPage.java

```diff
@@ -59,20 +74,168 @@ public EventFieldEditingPage(IField field) {
-		setDescription(MESSAGE_PRESET_MANAGER_PAGE_DESCRIPTION);
+		setTitle(MESSAGE_FIELD_EDITING_PAGE_TITLE);
+		setDescription(MESSAGE_FIELD_EDITING_PAGE_DESCRIPTION);
 
 		ScrolledComposite sc = new ScrolledComposite(parent, SWT.H_SCROLL | SWT.V_SCROLL);
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
 
-		// TODO: create field editor control here
-		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create field editor control here");
+		container.setLayout(new GridLayout());
+		container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+
+		Composite gridSection = new Composite(container, SWT.NONE);
+		gridSection.setLayout(new GridLayout(2, false));
+
+		createPageContent(gridSection);
+		gridSection.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+
+		popuateStoredValues();
+		bindListeners();
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
 	}
+
+	private void createPageContent(Composite parent) {
+		GridData gdLabel = new GridData(SWT.BEGINNING, SWT.CENTER, false, false);
+		GridData gdText = new GridData(SWT.FILL, SWT.CENTER, true, true);
+		gdText.minimumWidth = 0;
+		gdText.widthHint = 300;
+
+		Label label = createLabel(parent, "Name:");
+		label.setLayoutData(gdLabel);
+
+		name = createText(parent);
+		name.setMessage("Field Name");
+		name.setLayoutData(gdText);
+
+		label = createLabel(parent, "Expression:");
+		label.setLayoutData(gdLabel);
+
+		expression = createText(parent);
+		expression.setMessage("Expression to capture");
+		expression.setLayoutData(gdText);
+
+		label = createLabel(parent, "Description:");
+		label.setLayoutData(gdLabel);
+
+		GridData largeText = new GridData(SWT.FILL, SWT.CENTER, true, true);
+		largeText.minimumWidth = 0;
+		largeText.widthHint = 300;
+		largeText.heightHint = 100;
+		largeText.horizontalSpan = 2;
+
+		description = createTextMulti(parent);
+		description.setMessage("Description of field");
+		description.setLayoutData(largeText);
+
+		label = createLabel(parent, "Content Type:");
+		label.setLayoutData(gdLabel);
+
+		contentType = new Combo(parent, SWT.DROP_DOWN | SWT.READ_ONLY);
+		List<String> contentTypes = new ArrayList<>();
+		for (ICapturedValue.ContentType type : IMethodParameter.ContentType.values()) {
+			contentTypes.add(type.name());
+		}
+		contentType.setItems(contentTypes.toArray(new String[0]));
+		contentType.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+
+		label = createLabel(parent, "Relational Key:");
+		label.setLayoutData(gdLabel);
+
+		relationalKey = createText(parent);
+		relationalKey.setMessage("schema://some-uri");
+		relationalKey.setLayoutData(gdText);
+
+		label = createLabel(parent, "Converter Type:");
+		label.setLayoutData(gdLabel);
+
+		converterType = createText(parent);
+		converterType.setMessage("com.company.project.MyConverter");
```

Comment:
String constants as static class variables.

Reply:
... but not localizable

Reply:
what about making default `org.openjdk.jmc.agent.converters.DefaultStringConverter`?

Reply:
On second thought: no, the default should just be empty so that no converter is used.